app/channel.go

```diff
@@ -79,10 +78,10 @@ func (a *App) JoinDefaultChannels(teamId string, user *model.User, shouldBeAdmin
 			}
 
 			if cmResult := <-a.Srv.Store.Channel().SaveMember(cm); cmResult.Err != nil {
-				err = cmResult.Err
+				returnErr = cmResult.Err
 			}
-			if result := <-a.Srv.Store.ChannelMemberHistory().LogJoinEvent(user.Id, channel.Id, model.GetMillis()); result.Err != nil {
-				mlog.Warn(fmt.Sprintf("Failed to update ChannelMemberHistory table %v", result.Err))
+			if resultEvent := <-a.Srv.Store.ChannelMemberHistory().LogJoinEvent(user.Id, channel.Id, model.GetMillis()); resultEvent.Err != nil {
```

Comment:
I commented to this effect over at https://github.com/mattermost/mattermost-server/pull/10001#discussion_r242667710, but I think `go vet` is going too far here, and that we should consider allowing shadowing within conditional variables. I don't thin renaming `result` to `resultEvent` meaningfully increases visibility here, especially since if we want to use `resultEvent` outside of the block, we're likely to avoid the conditional declaration in the first place and then would deal with the shadowing organically.

Reply:
You correct. `resultEvent` doesn't provide any additional information.

I would like to point out that using new names for `result` is something common in the codebase like in https://github.com/mattermost/mattermost-server/blob/85a344ae93a0249e4edc3ba557e650059bb454be/app/channel.go#L80.